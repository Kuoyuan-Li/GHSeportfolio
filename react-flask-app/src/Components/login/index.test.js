import React from 'react'
import Index from './index'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';

describe ('Index page test', () => {
    test('redirect to login page when click login', () => {
        const historyMock = { push: jest.fn() }
        const wrapper = shallow(<Index history = {historyMock}/>)
        wrapper.find('button').at(0).simulate('click');       
        expect(historyMock.push.mock.calls[0]).toEqual(['/login']);
    });

    test('redirect to register page when click register', () => {
        const historyMock = { push: jest.fn() }
        const wrapper = shallow(<Index history = {historyMock}/>)
        wrapper.find('button').at(1).simulate('click');       
        expect(historyMock.push.mock.calls[0]).toEqual(['/register']);
    });

    /*test('renders correctly', () => {
        const tree = renderer.create(<Index />).toJSON();
        expect(tree).toMatchSnapshot();
      });*/
    

});