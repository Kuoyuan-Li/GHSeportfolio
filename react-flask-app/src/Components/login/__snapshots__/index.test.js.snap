// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page test renders correctly 1`] = `
<div
  id="login"
>
  <div
    class="container"
  >
    <div
      class="col-md-12"
    >
      <h3>
        Eportfolio System
      </h3>
      <div
        class="col-md-12"
      >
        <button
          class="button button1"
          onClick={[Function]}
        >
          Log in
        </button>
      </div>
      <div
        class="col-md-12"
      >
        <button
          class="button button2"
          onClick={[Function]}
        >
          New user? Register Now!
        </button>
      </div>
    </div>
  </div>
</div>
`;
